src/Console/ConfigurationResolver.php

```diff
@@ -702,13 +702,32 @@ private function validateRules(array $rules)
         );
 
         if (count($unknownFixers)) {
-            throw new InvalidConfigurationException(sprintf(
-                'The rules contain unknown fixers (%s).',
-                implode(', ', $unknownFixers)
-            ));
+            $message = 'The rules contain unknown fixers (';
+            foreach ($unknownFixers as $unknownFixer) {
+                $alternative = $this->getAlternativeRule($unknownFixer, $availableFixers);
+                $message .= sprintf(
+                    '"%s"%s, ',
+                    $unknownFixer,
+                    null === $alternative ? '' : ' maybe use "'.$alternative.'"'
+                );
+            }
+
+            throw new InvalidConfigurationException(substr($message, 0, -2).').');
         }
     }
 
+    private function getAlternativeRule($unknownFixer, array $availableFixers)
+    {
+        foreach ($availableFixers as $availableFixer) {
+            $distance = levenshtein($unknownFixer, $availableFixer);
+            if ($distance < 3) {
+                return $availableFixer;
```

Comment:
what if there are multiple candidates matching `< 3` ? we shall pick most similar

Reply:
Together with the DRY comment; I followed the version of the `describe` command.
I can cook something up that fixes this at both places, but I wanted to prevent to introduce a new `util` class kind of thing.

Reply:
```php
final class LevenshteinMatcher // go ahead and propose better
{
    /**
     * @var string[]
     */
    private $dict;

    public __construct(array $dict)
    {
        $this->dict = $dict;
    }

    public function match(string $value): string
    {
        // magic here
    }
}
```

```php
$matcher = new LevenshteinMatcher($availableFixers);
$bestMatch = $matcher->match($unknownFixer);
```

Reply:
maybe https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3113 ?